pkg/controllers/dbaddnode_reconcile.go

```diff
@@ -116,36 +110,37 @@ func (d *DBAddNodeReconciler) runAddNode(ctx context.Context, sc *vapi.Subcluste
 		if err := cleanupLocalFiles(ctx, d.Vdb, d.PRunner, pod.name); err != nil {
 			return ctrl.Result{}, err
 		}
+	}
 
-		debugDumpAdmintoolsConf(ctx, d.PRunner, atPod.name)
+	debugDumpAdmintoolsConf(ctx, d.PRunner, atPod.name)
 
-		if stdout, err := d.runAddNodeForPod(ctx, pod, atPod); err != nil {
-			// If we reached the node limit according to the license, end this
-			// reconcile successfully. We don't want to fail and requeue because
-			// this isn't going to get fixed until someone manually adds a new
-			// license.
-			if isLicenseLimitError(stdout) {
-				return ctrl.Result{}, nil
-			}
-			return ctrl.Result{}, err
+	if stdout, err := d.runAddNodeForPod(ctx, pods, atPod); err != nil {
+		// If we reached the node limit according to the license, end this
+		// reconcile successfully. We don't want to fail and requeue because
+		// this isn't going to get fixed until someone manually adds a new
+		// license.
+		if isLicenseLimitError(stdout) {
+			return ctrl.Result{}, nil
 		}
+		return ctrl.Result{}, err
+	}
 
-		debugDumpAdmintoolsConf(ctx, d.PRunner, atPod.name)
+	debugDumpAdmintoolsConf(ctx, d.PRunner, atPod.name)
 
-		// Invalidate the cached pod facts now that some pods have a DB now.
-		d.PFacts.Invalidate()
-	}
-	err := d.rebalanceShards(ctx, atPod, sc.Name)
-	return ctrl.Result{}, err
+	// Invalidate the cached pod facts now that some pods have a DB now.
+	d.PFacts.Invalidate()
```

Comment:
Do you mind telling me what this part means?

Reply:
The Invalidate() call will mark the podfacts as being stale.  The next reconciler (RebalanceShards) needs update pod facts to properly work since there are no new nodes that were not yet accounted for in the pod facts.